proposed/http-client/http-client.md

```diff
@@ -73,17 +73,14 @@ use Psr\Http\Message\RequestInterface;
 use Psr\Http\Client\Exception;
 
 /**
- * Exception for when a request failed, providing access to the failed request.
- *
- * This could be due to an invalid request, example when you can seek in the request body stream
```

Comment:
I want to give an example when to use this. 
1) when the request is invalid (missing method or other invalid config.)
2) runtime things, like when you **cant** seek in the request body.